root/etc/e-smith/events/actions/nethserver-dc-user-modify

```diff
@@ -40,3 +45,31 @@ if ($? != 0) {
     die("[ERROR] User $userName modification failed\n");
 }
 
+
+#Create a ldif file and use it to modify the shell access
+#a random name and a chmod is done for security
+my $path = '/var/lib/machines/nsdc/var/lib/samba/private/';
+my $fileName = "$userName-". sprintf("%08X", rand(0xFFFFFFFF)).".ldif";
+my $filePath = $path . $fileName;
+
+my $contentFile = qq (
+dn: cn=$userName,cn=Users,$baseDN
+changetype: modify
+replace: loginShell
+loginShell: $shell
+);
+
+open(my $fh, '>', $filePath) or die "Could not open file '$filePath' $!";
+chmod 0600, $filePath;
+print $fh $contentFile;
+close $fh;
+
+#Launch ldbmodify and expect the return
+system(qq(expect -c 'spawn -noecho /usr/bin/systemd-run -M nsdc -q -t /usr/bin/ldbmodify -H /var/lib/samba/private/sam.ldb /var/lib/samba/private/$fileName;
```

Comment:
You can avoid the temporary file management: here `ldbmodify` can read data from STDIN (tty); `expect` can `send` `$contentFile` to it directly

Reply:
Ok will look further

Reply:
I must take back what I've said: `ldbmodify` does not prompt for input, so it's difficult for `expect` to interact with it... If I come to a solution I'll let you know!